src/main/java/randoop/contract/EqualsHashcode.java

```diff
@@ -20,7 +20,10 @@ public boolean evaluate(Object... objects) {
     Object o1 = objects[0];
     Object o2 = objects[1];
 
-    return !o1.equals(o2) || o1.hashCode() == o2.hashCode();
+    if (o1.equals(o2)) {
+      return o1.hashCode() == o2.hashCode();
+    }
+    return true;
```

Comment:
I'm unfamiliar with this part of Randoop.  Why isn't the last changed line return false?

Reply:
The Javadoc of `evaluate` says:

> Returns true if the contract was satisfied or was not applicable.

If two objects are `equals` to one another, their hash codes must be equal.
If two objects are not `equals` to one another, then there is no further requirement.